3rdparty/bx/include/bx/platform.h

```diff
@@ -133,8 +133,14 @@
 #endif //
 
 #if BX_CPU_PPC
-#	undef  BX_CPU_ENDIAN_BIG
-#	define BX_CPU_ENDIAN_BIG 1
+// _LITTLE_ENDIAN exists on ppc64le.
```

Comment:
I would suggest two changes:
* use # if defined ... rather than # if ... to avoid warning when build on ppc64 (BE)
* replace test of  _LITTLE_ENDIAN by __BIG_ENDIAN__ gcc predefined macro, to keep default as little endian.
```
 #if BX_CPU_PPC
-#      undef  BX_CPU_ENDIAN_BIG
-#      define BX_CPU_ENDIAN_BIG 1
+// __BIG_ENDIAN__ is gcc predefined macro
+#      if defined(__BIG_ENDIAN__)
+#              undef  BX_CPU_ENDIAN_BIG
+#              define BX_CPU_ENDIAN_BIG 1
+#      else
+#              undef  BX_CPU_ENDIAN_LITTLE
+#              define BX_CPU_ENDIAN_LITTLE 1
+#      endif
 #else
 #      undef  BX_CPU_ENDIAN_LITTLE
 #      define BX_CPU_ENDIAN_LITTLE 1
```

Reply:
Not to come off as too cranky, but I would suggest suggesting your suggestions upstream to @bkaradzic, as the topmost subdirectory in which that file is located, "3rdparty", would imply exactly that - that these are third-party libraries that we do not ourselves maintain, nor have much desire to fork from their mainline repositories.

Reply:
as suggested just submitted to bx as https://github.com/bkaradzic/bx/pull/224

Reply:
Accepted.